app/controllers/distributions_controller.rb

```diff
@@ -58,6 +58,7 @@ def create
   def new
     @distribution = Distribution.new
     @distribution.line_items.build
+    copy_from_donation
```

Comment:
My spidey-sense says that for this one, it might be better to:

1. Move `copy_from_donation` to the `Distribution` model, like you did with `copy_line_items`, it would accept a `:donation_id` and `:storage_location_id` as params
2. Invoke that method here in the controller like: `@distribution.copy_from_donation(params[:donation_id], params[:storage_location_id])`  (might need to check for `nil?` first? or at least handle cases where either is not set)

As it stands, it feels like the Controller is doing logic that should be encapsulated in the model